src/main/java/de/tum/in/www1/artemis/repository/LearningPathRepository.java

```diff
@@ -47,12 +45,24 @@ SELECT COUNT (learningPath)
             """)
     long countLearningPathsOfEnrolledStudentsInCourse(@Param("courseId") long courseId);
 
-    @EntityGraph(type = LOAD, attributePaths = { "competencies", "competencies.lectureUnits", "competencies.lectureUnits.completedUsers", "competencies.exercises",
-            "competencies.exercises.studentParticipations" })
-    Optional<LearningPath> findWithEagerCompetenciesAndLearningObjectsAndCompletedUsersById(long learningPathId);
+    @Query("""
+                SELECT learningPath
+                FROM LearningPath learningPath
+                LEFT JOIN FETCH learningPath.competencies competencies
```

Comment:
Indentation would be nice
How often do we expect this query to be executed because it fetches a lot of things.